resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/SslConfigurationProvider.scala

```diff
@@ -52,51 +61,73 @@ private[spark] class SslConfigurationProvider(
   private val sslSecretsDirectory = s"$DRIVER_CONTAINER_SECRETS_BASE_DIR/$kubernetesAppId-ssl"
 
   def getSslConfiguration(): SslConfiguration = {
-    val (driverSubmitSslOptions, isKeyStoreLocalFile) = parseDriverSubmitSslOptions()
-    if (driverSubmitSslOptions.enabled) {
+    val driverSubmitSslOptions = parseDriverSubmitSslOptions()
+    if (driverSubmitSslOptions.storeBasedSslOptions.enabled) {
       val sslSecretsMap = mutable.HashMap[String, String]()
+      val storeBasedSslOptions = driverSubmitSslOptions.storeBasedSslOptions
       val sslEnvs = mutable.Buffer[EnvVar]()
```

Comment:
Can you split setting on sslEnvs and sslSecretsMap. You can just concat the transformed options which is more in line with what scala does. Then you don't need mutable buffers